modules/objc/examples/code_snippets/SampleCodeTest.m

```diff
@@ -1368,1143 +1376,1271 @@ - (void) dontTestListenerCertificateAuthenticatorCallback {
+- (void)disconnect;
+- (void)receive:(NSData*)data;
+@end
+
+@implementation ActivePeer
+
+- (instancetype) init {
+    self = [super init];
+    if (self) {
+        NSError* error = nil;
+        // tag::message-endpoint[]
+        CBLDatabase *database = [[CBLDatabase alloc] initWithName:@"dbname" error: &error];
+
+        // The delegate must implement the `CBLMessageEndpointDelegate` protocol.
+        NSString* id = @"";
+        CBLMessageEndpoint *endpoint =
+        [[CBLMessageEndpoint alloc] initWithUID:@"UID:123"
+                                         target:id
+                                   protocolType:kCBLProtocolTypeMessageStream
+                                       delegate:self];
+        // end::message-endpoint[]
 
-    func mystopfunc() {
-// tag::p2p-act-rep-stop[]
-    // Remove the change listener
-    [thisReplicator removeChangeListenerWithToken: thisListenerToken];
+        // tag::message-endpoint-replicator[]
+        CBLReplicatorConfiguration *config =
+        [[CBLReplicatorConfiguration alloc] initWithDatabase:database target: endpoint];
 
-    // Stop the replicator
-    [thisReplicator stop];
-// end::p2p-act-rep-stop[]
+        // Create the replicator object.
+        CBLReplicator *replicator = [[CBLReplicator alloc] initWithConfig: config];
+        [replicator start];
+        // end::message-endpoint-replicator[]
+    }
+    return self;
 }
 
-// Additional Snippets from above
-    // tag::p2p-act-rep-config-cacert[]
-    // Configure Server Security -- only accept CA Certs
-    thisConfig.acceptOnlySelfSignedServerCertificate = NO; // <.>
-
-    // end::p2p-act-rep-config-cacert[]
+// tag::create-connection[]
+- (id<CBLMessageEndpointConnection>)createConnectionForEndpoint:(CBLMessageEndpoint *)endpoint {
+    return [[ActivePeerConnection alloc] init];
+}
+// end::create-connection[]
 
+@end
 
-    // tag::p2p-act-rep-config-pinnedcert[]
-    // Return the remote pinned cert (the listener's cert)
-    thisConfig.pinnedServerCertificate = thisCert; // Get listener cert if pinned
 
-    // end::p2p-act-rep-config-pinnedcert[]
+@implementation ActivePeerConnection {
+    id <CBLReplicatorConnection> _replicatorConnection;
+}
 
+- (void)disconnect {
+    // tag::active-replicator-close[]
+    [_replicatorConnection close:nil];
+    // end::active-replicator-close[]
+}
 
+// tag::active-peer-open[]
+/* implementation of CBLMessageEndpointConnection */
+- (void)open:(nonnull id<CBLReplicatorConnection>)connection completion:(nonnull void (^)(BOOL, CBLMessagingError * _Nullable))completion {
+    _replicatorConnection = connection;
+    completion(YES, nil);
+}
+// end::active-peer-open[]
 
+// tag::active-peer-send[]
+/* implementation of CBLMessageEndpointConnection */
+- (void)send:(nonnull CBLMessage *)message completion:(nonnull void (^)(BOOL, CBLMessagingError * _Nullable))completion {
+    NSData *data = [message toData];
+    NSLog(@"%@", data);
```

Comment:
Can we remove NSLog here?

Reply:
This one is to demo, how the data can be accessed (if needed can be send it to the other peer).